Sources/Kafka/Configuration/KafkaConfiguration+Metrics.swift

```diff
@@ -0,0 +1,133 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-client open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-kafka-client project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-client project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Metrics
+
+extension KafkaConfiguration {
+    // MARK: - Metrics
+
+    /// Use to configure metrics.
+    public struct KafkaMetrics: Sendable {
+        internal var someMetricsSet: Bool {
+            self.timestamp != nil ||
+                self.time != nil ||
+                self.age != nil ||
+                self.replyQueue != nil ||
+                self.messageCount != nil ||
+                self.messageSize != nil ||
+                self.messageMax != nil ||
+                self.messageSizeMax != nil ||
+                self.totalRequestsSent != nil ||
+                self.totalBytesSent != nil ||
+                self.totalResponsesRecieved != nil ||
+                self.totalBytesReceived != nil ||
+                self.totalMessagesSent != nil ||
+                self.totalMessagesBytesSent != nil ||
+                self.totalBytesReceived != nil ||
+                self.metadataCacheCount != nil
+        }
+
+        /// librdkafka's internal monotonic clock (microseconds)
+        public var timestamp: Gauge?
+        /// Wall clock time in seconds since the epoch
+        public var time: Timer?
+        /// Time since this client instance was created
+        public var age: Timer?
+        /// Number of ops (callbacks, events, etc) waiting in queue for application to serve
+        public var replyQueue: Gauge?
+        /// Current number of messages in producer queues
+        public var messageCount: Gauge?
+        /// Current total size of messages in producer queues
+        public var messageSize: Gauge?
+        /// Threshold: maximum number of messages allowed allowed on the producer queues
+        public var messageMax: Gauge?
+        /// Threshold: maximum total size of messages allowed on the producer queues
+        public var messageSizeMax: Gauge?
+
+        /// Total number of requests sent to Kafka brokers
+        public var totalRequestsSent: Gauge?
```

Comment:
If this is a total then it shouldn't be a `Gauge` but rather a `Counter` or are we using a `Gauge` because we only get the latest values and just have to set it?
```suggestion
        /// Total number of requests sent to Kafka brokers.
        public var totalKafkaBrokerRequests: Counter?
```

Reply:
I agree that it should be a counter but your guess is right - we receive the latest values for this field. Therefore, using gauge here.